### PR TITLE
refactor: fixing sonar high founding

### PR DESCRIPTION
**Pull request by podlesrafal**

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/idp/impl/IdentityProviderManagerImpl.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ public void onEvent(Event<IdentityProviderEvent, Payload> event) {
                     removeIdentityProvider(event.content().getId());
                     gatewayMetricProvider.decrementIdp();
                     break;
+                default: break;
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/idp/impl/IdentityProviderManagerImpl.java:142]: break in default? 

Can we not use a new switch here?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/client/impl/ClientManagerImpl.java [hunk 2]
```diff
@@ -97,6 +97,7 @@ public void onEvent(Event<ApplicationEvent, Payload> event) {
                     removeClient(event.content().getId());
                     gatewayMetricProvider.decrementApp();
                     break;
+                default: break;
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/client/impl/ClientManagerImpl.java:100]: break in default?

Can we not use a new switch here?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/granter/CompositeTokenGranter.java [hunk 3]
```diff
@@ -103,8 +103,6 @@ public class CompositeTokenGranter implements TokenGranter, InitializingBean {
     @Autowired
     private AuditService auditService;
 
-    public CompositeTokenGranter() { }
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/granter/CompositeTokenGranter.java:106]: Are you sure it is not needed to initialize bean?

File: gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java [hunk 4]
```diff
@@ -311,6 +308,8 @@ protected void generateJWTProcessor(OpenIDConnectIdentityProviderConfiguration c
                         keyProcessor = new HMACKeyProcessor();
                         keyProcessor.setJwkSourceResolver(new MACJWKSourceResolver(resolverParameter));
                         break;
+                    default:
```
Comment (mstasgravitee) [inline gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java:311]: break in default?

Can we not use a new switch here?

File: gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java [hunk 5]
```diff
@@ -267,6 +264,8 @@ private void synchronizeDomain(Event event) {
             case DELETE:
                 securityDomainManager.undeploy(domainId);
                 break;
+            default:
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java:267]: break in default?

Can we not use a new switch here?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/granter/CompositeTokenGranter.java (hunk 3 above)
Comment (mregulski-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/granter/CompositeTokenGranter.java:106]: If there's no other constructor, java creates a no-arg constrcutor implicitly, so there's no need to declare it.

File: gravitee-am-common/src/main/java/io/gravitee/am/common/event/AlertNotifierEvent.java [hunk 6]
```diff
@@ -26,18 +26,11 @@ public enum AlertNotifierEvent {
...
-            case UPDATE:
-                notifier = AlertNotifierEvent.UPDATE;
-                break;
-            case DELETE:
-                notifier = AlertNotifierEvent.UNDEPLOY;
-                break;
-        }
-        return notifier;
+        return switch (action) {
+            case CREATE -> AlertNotifierEvent.DEPLOY;
```
Comment (mregulski-gravitee) [inline gravitee-am-common/src/main/java/io/gravitee/am/common/event/AlertNotifierEvent.java:30]: Shouldn't `BULK_*` actions be mapped to the same `AlertNotifierEvents` as non-bulk ones?
(Same applies to other `Action` -> enum mappers)

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/idp/impl/IdentityProviderManagerImpl.java (hunk 1 above)
Comment (podlesrafal) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/idp/impl/IdentityProviderManagerImpl.java:142]: https://sonarcloud.io/organizations/gravitee-io/rules?open=java%3AS131&rule_key=java%3AS131

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/client/impl/ClientManagerImpl.java (hunk 2 above)
Comment (podlesrafal) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/client/impl/ClientManagerImpl.java:100]: https://sonarcloud.io/organizations/gravitee-io/rules?open=java%3AS131&rule_key=java%3AS131

File: gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java (hunk 4 above)
Comment (podlesrafal) [reply gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java:311]: https://sonarcloud.io/organizations/gravitee-io/rules?open=java%3AS131&rule_key=java%3AS131

File: gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java (hunk 5 above)
Comment (podlesrafal) [reply gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java:267]: https://sonarcloud.io/organizations/gravitee-io/rules?open=java%3AS131&rule_key=java%3AS131

**Review by mstasgravitee — APPROVED** [review]

File: gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractSocialAuthenticationProvider.java [hunk 7]
```diff
@@ -128,9 +134,9 @@ private boolean roleMappingEnabled() {
     }
 
     protected final class Token {
```
Comment (mregulski-gravitee) [inline gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractSocialAuthenticationProvider.java:136]: could be converted to a `record` or `@lombok.Value`

File: gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java [hunk 8]
```diff
@@ -211,24 +210,22 @@ private void deployDomains() {
         List<Domain> domains = findDomains.blockingGet();
 
         // deploy security domains
-        domains.stream().forEach(domain -> securityDomainManager.deploy(domain));
+        domains.forEach(domain -> securityDomainManager.deploy(domain));
```
Comment (mregulski-gravitee) [inline gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java:213]:
```
domains.forEach(securityDomainManager::deploy)
```

**Review by mregulski-gravitee — APPROVED** [review]

**Pull request merged**